packages/slice-machine/components/Grid/index.tsx

```diff
@@ -1,30 +1,35 @@
+import React from "react";
 import { Box } from "theme-ui";
 
-const Grid = ({
+interface GridProps<T> {
+  elems: readonly (T | undefined)[];
+  renderElem: (elem: T, index: number) => JSX.Element | null;
+  gridTemplateMinPx?: string;
+}
+
+function Grid<T extends unknown>({
   elems,
   gridTemplateMinPx = "320px",
   renderElem,
-}: {
-  elems: any;
-  gridTemplateMinPx?: string;
-  renderElem: Function;
-}) => {
+}: GridProps<T>): JSX.Element {
   return (
     <Box
       as="section"
       sx={{
         display: "grid",
         gridTemplateColumns: `repeat(auto-fill, minmax(${gridTemplateMinPx}, 1fr))`,
-        gridGap: "16px" as any,
+        gridGap: "16px",
         pt: 2,
         mb: 3,
       }}
     >
-      {elems.map((elem: any, i: number) => (
-        <span key={`list-item-${i + 1}`}>{renderElem(elem, i)}</span>
-      ))}
+      {elems.map((elem: T | undefined, i: number) =>
+        elem ? (
+          <span key={`list-item-${i + 1}`}>{renderElem(elem, i)}</span>
```

Comment:
pretty sure when removing / sorting items, these indexes can get mixed up. I've got no idea how to handle it when dealing with generic list items.

Reply:
i'm not sure to understand.

Reply:
ah yes i see what you mean, having the key made with the element and the index so that it's really unique and will re-render if the items changed position. Good catch 👍